services/brig/src/Brig/Data/User.hs

```diff
@@ -404,9 +406,18 @@ lookupRichInfoMultiUsers users = do
 -- successful login.
 lookupUserTeam :: UserId -> AppIO (Maybe TeamId)
 lookupUserTeam u =
-  join . fmap runIdentity
+  (runIdentity =<<)
     <$> retry x1 (query1 teamSelect (params LocalQuorum (Identity u)))
 
+lookupUserUnverifiedEmail :: MonadClient m => UserId -> m (Maybe Email)
+lookupUserUnverifiedEmail userId =
+  (runIdentity =<<) <$> retry x1 (query1 cql (params LocalQuorum (Identity $ toUUID userId)))
+  where
+    cql :: PrepQuery R (Identity UUID) (Identity (Maybe Email))
```

Comment:
I think you could still use `UserId` here, but it's not important, UUID is also fine.